doc/doxygen/ucxdox

```diff
@@ -787,11 +790,14 @@ INPUT_ENCODING         = UTF-8
 # *.md, *.mm, *.dox, *.py, *.f90, *.f, *.for, *.tcl, *.vhd, *.vhdl, *.ucf,
 # *.qsf, *.as and *.js.
 
-FILE_PATTERNS          = ucp.h      \
-                         ucp_def.h  \
-                         uct.h      \
-                         uct_def.h  \
-                         status.h   \
+FILE_PATTERNS          = ucp.h          \
+                         ucp_def.h      \
+                         uct.h          \
+                         uct_def.h      \
+                         status.h       \
+                         async_fwd.h    \
+                         compiler_def.h \
+                         time_def.h     \
```

Comment:
@shamisp I added these three files (time_def.h, compiler_def.h, async_fwd.h) to the documentation index because they are in public interface. I think all functions/types should be well documented from these files.